### PR TITLE
Maint/20 remove duplicate mpi implementation

### PR DESCRIPTION
**Pull request by AntoineSIMTEK**
I removed obscure and deprecated helper files remaining from the development process. Tools for profiling LBFGS remain in the helpers, since they are useful templates. I will create  a separate PR for PEP8 enforcement soon.

### CODE REVIEW

File: tests/test_LBFGS_minimization_problems.py [hunk 1]
```diff
@@ -27,22 +27,14 @@
...
 import scipy.optimize
-from NuMPI.Optimization.LBFGS_Matrix_H import LBFGS
+from NuMPI.Optimization import LBFGS
+from NuMPI import MPI
 import tests.minimization_problems as mp
 
 import pytest
-@pytest.mark.parametrize("Objective",[mp.Extended_Rosenbrock]) # Only these where Solution is analytically known
-@pytest.mark.parametrize("n",[10,20])
-def test_analytical_min(Objective,n):
```
Comment (pastewka) [inline tests/test_LBFGS_minimization_problems.py:36]: Why did you delete this test?

Comment (AntoineSIMTEK) [reply tests/test_LBFGS_minimization_problems.py:36]: because it is a duplicate of test_MPI_LBFGS:test_analytical_min

Up to the number of gridpoints however. I can try to add n=10 and 20 as fixture.

Also the LBFGS tests are split into a lot of different files, maybe I can get that bettter organized.

Comment (pastewka) [reply tests/test_LBFGS_minimization_problems.py:36]: Okay! Please merge this PR

**Comment by AntoineSIMTEK** [comment]
@pastewka , I changed the test directory structure.
